engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/execution/RuntimeExecutionHooks.java

```diff
@@ -0,0 +1,31 @@
+package org.enso.interpreter.instrument.execution;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class RuntimeExecutionHooks implements ExecutionHooks {
+
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+  private final List<Runnable> hooks = new ArrayList<>();
+
+  public RuntimeExecutionHooks() {}
+
+  @Override
+  public void add(Runnable hook) {
+    hooks.add(hook);
```

Comment:
Should probably verify that we don't run the same hooks

Reply:
Good idea :+1: